src/Pagination/ConnectionField.php

```diff
@@ -85,6 +85,10 @@ public function edgeResolver(Paginator $paginator, array $args, GraphQLContext $
                         $data['node'] = $item;
                         break;
 
+                    case 'pivot':
+                        $data['pivot'] = $item->pivot->getAttributes();
+                        break;
+
                     default:
                         // All other fields on the return type are assumed to be part
                         // of the edge, so we try to locate them in the pivot attribute
```

Comment:
No, the current logic assumes that the pivot properties must be on the edge type itself, i.e.:
````
{
  edges {
    cursor
    node {
      ...
    }
    pivotField -> internally points to node->pivot->pivotField
    pivot -> internally points to node->pivot->pivot which is obviously wrong
  }
}
````

However, this change would make it possible to have a pivot field on the edge that separates the pivot type from the edge type:
````
{
  edges {
    cursor
    node {
      ...
    }
    pivot {
       ...
    } -> internally points to node->pivot
  }
}
````

This separation is especially handy on the client-side because we can update the pivot type using graphcache during mutations without actually affecting the edge type.

I hope this makes some sense :blush:

Reply:
I can see the benefit and am fine to go ahead with this change. Please add tests, documentation and a changelog entry.